README.md

```diff
@@ -251,6 +254,30 @@ Future.node(done => fs.readFile('package.json', 'utf8', done))
 //> "{...}"
 ```
 
+#### loop
+##### `.loop :: (b -> Future a (Iteration b)) -> b -> Future a b`
+
+Stack- and memory-safe asynchronous "recursion" based on [FantasyLand ChainRec][26].
+Calls the given function with the initial value, and expects a Future of an
+[Iteration](#type-signatures). If the Iteration is incomplete (`{done: false}`),
+then the function is called again with the Iteration value until it returns a
+Future of a complete (`{done: true}`) Iteration.
+
+```js
+const Next = x => ({done: false, value: x});
+const Done = x => ({done: true, value: x});
+Future.loop((x => Future.of(x < 1000000 ? Next(x + 1) : Done(x))), 0)
+.fork(console.error, console.log);
+//> 1000000
+```
+
+#### chainRec
+##### `.chainRec :: ((((b -> Next b), (b -> Done b), b) -> Future a (Iteration b)), b) -> Future a b`
```

Comment:
type of initial value and type of value passed to `Next` constructor should be same, but constructor of `Done` might take value of different type, so signature is not quite correct.

For example:
```js
const initialPage =  { next : 'posts/1' , posts:[]}
const lastPagePostLikes = Future.chainRec(
  (next, done, page) => (page.next
    ? HTTP.fetchJSON(page.next).map(next)
    : Task.of(done(page.posts.map(getLikes).sum()))
  ), initialPage
)
```
here type of accumulator is `Page` but result is `Number`.
and more explicit type of `chainRec` in that case would be:
```hs
chainRec :: ((Page -> Next Page, Number -> Done Number, Page) -> Future r (Iteration Page Number), Page) -> Future r Number
```
where `Iteration a b = Next a | Done b`.

so general `chainRec` should be something like this:
```hs
chainRec :: ((a -> Next a, b -> Done b, a) -> Future r (Iteration a b), a) -> Future r b
```

also as `Iteration`, `Next` and `Done` are just internal thing it could also be like this:
```hs
chainRec :: ((a -> Iteration a b, b -> Iteration a b, a) -> Future r (Iteration a b), a) -> Future r b
```
and then like this
```hs
chainRec :: ((a -> c, b -> c, a) -> Future r c, a) -> Future r b
```